libs/newsletters-data-client/src/lib/schemas/data-collection-schema.ts

```diff
@@ -0,0 +1,84 @@
+import { z } from 'zod';
+import { nonEmptyString, urlPathString } from '../zod-helpers';
+import {
+	newsletterDataSchema,
+	onlineArticleSchema,
+	regionFocusEnumSchema,
+} from './newsletter-data-type';
+import {
+	automatedFrontSectionSchema,
+	readMoreSectionSchema,
+	renderingOptionsSchema,
+} from './rendering-options-data-type';
+
+// TO DO - can we fetch the list of supported functions from email-rendering
+// and dynamically create the schema? is that overkill?
+// NOTE - data validation schema types partBuilderFunction as non-empty string
+// seemed too rigid to use the names in the data schema - might need a data migration
+// to add/remove/deprecate options. Defining the options for data collection only
+// give enough flexibility
+const builderFunctionNames = [
```

Comment:
This also makes me think we need to have a solution for display lables for this kind of content. Our forms all look a bit WIP because there is so much camelCase on display. Not one for this PR but an important health ticket.
There is an example somewhere of extending Zod schemas to add a label property that can be used for UI purposes

Reply:
I think it's set up so the Zod field's `description` property is used as the field label (if set - if not it falls back to the field Id). 
But, agreed, we probably could handle that better.